UI/js/signup.js

```diff
@@ -0,0 +1,267 @@
+    firstNameErrorMessage.classList.add('is-visible');
+    firstNameErrorMessage.innerHTML = errors.firstName;
+  } else {
+    delete (errors.firstName);
+  }
+
+
+  if (!lastName.value.trim()) {
+    errors.lastName = 'Last name is required';
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else if (lastName.value.trim() && !/^[A-Za-z]*$/.test(lastName.value.trim())) {
+    errors.lastName = `Last name must consist of only alphabets
+    and must contain no spaces between characters`;
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else if (lastName.value.trim() && lastName.value.trim().length > 50) {
+    errors.lastName = 'Last name must be fewer than 50 characters';
+    lastNameErrorMessage.classList.add('is-visible');
+    lastNameErrorMessage.innerHTML = errors.lastName;
+  } else {
+    delete (errors.lastName);
+  }
+
+
+  if (!email.value.trim()) {
+    errors.email = 'Email is required';
+    emailErrorMessage.classList.add('is-visible');
+    emailErrorMessage.innerHTML = errors.email;
+  } else if (email.value.trim() && (!(/^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$/).test(email.value.trim()))) {
+    errors.email = 'Email is invalid';
+    emailErrorMessage.classList.add('is-visible');
+    emailErrorMessage.innerHTML = errors.email;
+  } else {
+    delete (errors.email);
+  }
+
+  if (!password.value.trim()) {
+    errors.password = 'Password is required';
+    passwordErrorMessage.classList.add('is-visible');
+    passwordErrorMessage.innerHTML = errors.password;
+  } else if (password.value.trim() && password.value.trim().length < 8) {
+    errors.password = 'Password must contain between 8 - 20 characters';
+    passwordErrorMessage.classList.add('is-visible');
+    passwordErrorMessage.innerHTML = errors.password;
+  } else if (password.value.trim() && password.value.trim().length > 20) {
+    errors.password = 'Password must contain between 8  - 20 characters';
+    passwordErrorMessage.classList.add('is-visible');
+    passwordErrorMessage.innerHTML = errors.password;
+  } else {
+    delete (errors.password);
+  }
+
+  if (!confirmPassword.value.trim()) {
+    errors.confirmPassword = 'Re-enter password';
+    confirmPasswordErrorMessage.classList.add('is-visible');
+    confirmPasswordErrorMessage.innerHTML = errors.confirmPassword;
+  } else if (confirmPassword.value.trim()
+  && confirmPassword.value.trim() !== password.value.trim()) {
+    errors.confirmPassword = 'Passwords do not match';
+    confirmPasswordErrorMessage.classList.add('is-visible');
+    confirmPasswordErrorMessage.innerHTML = errors.confirmPassword;
+  } else {
+    delete (errors.confirmPassword);
+  }
+
+  if (Object.keys(errors).length > 0) {
+    signupButton.disabled = true;
+  } else {
+    signupButton.disabled = false;
+  }
+});
+
+firstName.addEventListener('blur', () => {
```

Comment:
A for-loop iterating through the list of events you want to listen to might help you avoid re-writing the checks for every event. Might help reduce code repetition.